presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestUnion.java

```diff
@@ -151,6 +180,13 @@ private void assertPlanIsFullyDistributed(Plan plan)
         assertEquals(gathers.size(), 1, "Only a single REMOTE GATHER was expected");
     }
 
+    private boolean shouldBeDistributed(PlanNode planNode)
+    {
+        return planNode instanceof AggregationNode ||
+                planNode instanceof JoinNode ||
+                (planNode instanceof TopNNode && ((TopNNode) planNode).getStep().equals(TopNNode.Step.PARTIAL));
```

Comment:
The method will grow (i once already wanted to add something here), so what about putting this in a more lengthy but more readable way? (Note, I also changed TopNNode's condition)

```
if (planNode instanceof JoinNode) {
    return true;
}

if (planNode instanceof AggregationNode) {
    return true; // here you could differentiate between partial/...
}

if (planNode instanceof TopNNode) {
    return ((TopNNode) planNode).getStep() != TopNNode.Step.FINAL;
}

return false;
```